### PR TITLE
Trim filtered table to original time range in TabOpLowPassFilter

### PR DESCRIPTION
**Pull request by nickbianco**
Fixes issue #3287

### Brief summary of changes
Updates `TabOpLowPassFilter` to trim the table back to the original time range.

### Testing I've completed
Added a unit test to `testTableProcessor.cpp`.

### CHANGELOG.md (choose one)

- updated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3969)

### CODE REVIEW

File: OpenSim/Simulation/Test/testTableProcessor.cpp [hunk 1]
```diff
@@ -76,3 +76,13 @@ TEST_CASE("TableProcessor") {
...
+
+TEST_CASE("TabOpLowPassFilter retains original time range") {
+    TimeSeriesTable table(std::vector<double>{1, 2, 3, 4, 5, 6},
+            SimTK::Test::randMatrix(6, 2), std::vector<std::string>{"a", "b"});
+    TableProcessor proc = TableProcessor(table) | TabOpLowPassFilter(6);
+    TimeSeriesTable out = proc.process();
+    CHECK(out.getNumRows() == 6);
+    CHECK(out.getIndependentColumn().front() == 1);
+    CHECK(out.getIndependentColumn().back() == 6);
+}
```
Comment (tkuchida) [inline OpenSim/Simulation/Test/testTableProcessor.cpp:88]: Nit: missing newline at eof

**Review by tkuchida — APPROVED** [review]

**Review by nickbianco — COMMENTED** [review]
Thanks @tkuchida! Requesting a quick re-review since I made some additional changes on top of the requested fix.

> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3969#-:-OC-rnT82iInDBG1TeZn:bpsmnci)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @tkuchida)

File: OpenSim/Simulation/TableProcessor.h [hunk 2]
```diff
@@ -183,19 +183,36 @@ class OSIMSIMULATION_API TabOpLowPassFilter : public TableOperator {
     OpenSim_DECLARE_PROPERTY(cutoff_frequency, double,
             "Low-pass cutoff frequency (Hz) (default is -1, which means no "
             "filtering).");
-    TabOpLowPassFilter() { constructProperty_cutoff_frequency(-1); }
+    OpenSim_DECLARE_PROPERTY(trim_to_original_time_range, bool,
+            "Trim the rows of the output table to match the original table's "
+            "time range after padding and filtering (default: true).");
```
Comment (tkuchida) [inline OpenSim/Simulation/TableProcessor.h:188]: Perhaps delete "after padding" as I don't think the user-facing documentation explains padding (what/why).

**Review by tkuchida — APPROVED** [review]
One optional suggestion 👇

**Comment by nickbianco** [comment]
Merging this. Thanks @tkuchida!

**Pull request merged**